rfcs/0000-shareable-courses.md

```diff
@@ -0,0 +1,150 @@
+* There should never be an assessment on a production instance that references a question that does not exist.
+
+The following facts and behaviors ensure those guarantees:
+
+* A leading `@` in a question ID in an assessment definition will unambiguously differentiate shareable course questions from course-local questions.
+
+* Once a course has been designated as shareable, it must remain shareable. That is, once a course with `"shareable": true` is synced to an instance of PrairieLearn, attempting to sync the same course with `"shareable": false` will be a sync error.
+
+* Once a question in a shareable course has been synced to an instance of PrairieLearn, attempting to perform a sync where the question would be renamed or deleted will be a sync error. *(Is there any special consideration necessary for local development?)*
+
+* When a course is synced, all questions that are referenced by its assessments will be checked. If a referenced shareable course is not available on that instance, or a question within a referenced shareable course does not exist, the sync will error. *(Note that this restriction is relaxed in local installations of PrairieLearn, where a user may not have the necessary permissions to clone the shareable course repository.)*
+
+## Access control
+
+Even though shareable courses are "courses", they will not be visible in PrairieLearn for students to register for.
+
+On a given PrairieLearn instance, the ability to sync a shareable course is controlled the same way that it is for courses: via the "course access roles" configuration.
+
+# Drawbacks
+
+## Inability to configure or augment questions
+
+With this proposed solution, there isn't an obvious way to control the behavior of a given question, such as adjusting the difficulty level or adjusting units/tolerances based on the needs of a particular course. There's also no way to augment questions, such as by providing additional instructions for a given question.
+
+It's unclear if either of these features would be needed in practice. However, it's worth calling them out to be aware of the limitations. These constraints might actually prove useful by encouraging sensible defaults and ensuring that questions are self-contained so that additional instructions/context aren't needed.
+
+# Alternatives
+
+## No formal support for sharing questions
+
+This is the current state of PrairieLearn. "Sharing" a question involves manually copying all of its files from one course to another. From a technical perspective, this solution is simple, as it requires no special treatment in PrairieLearn code. However, it scales poorly as it requires a lot of manual action to propagage question enhancements or fixes.
+
+## Share questions directly from courses
+
+Instead of having a special entity that stores shared questions, one could share questions directly from an existing course. From a technical perspective this is likely easier than the proposal of dedicated shareable courses.
```

Comment:
```suggestion
Instead of having a special flag to indicate a shareable course, one could share questions directly from any existing course. From a technical perspective this is slightly easier than the proposal of dedicated shareable courses.
```